sdks/python/apache_beam/options/pipeline_options.py

```diff
@@ -476,6 +499,26 @@ def _add_argparse_args(cls, parser):
         'time. NOTE: only supported with portable runners '
         '(including the DirectRunner)')
 
+  def validate(self, unused_validator):
+    errors = []
+    if beam.version.__version__ >= '3':
```

Comment:
I think when/if we reach to Beam version 3, we can remove this warning, and all and etc and and enable by default without an option to disable. So maybe we do not need this?

Reply:
This error is like a long-term TODO so that we don't forget to update defaults in v3.
It could mean that we enable everything and retire this flag. It depends on how stable type hints are.